data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3Source.java

```diff
@@ -47,7 +51,22 @@ public void start(Buffer<Record<Event>> buffer) {
         final ConfigBucketOwnerProviderFactory configBucketOwnerProviderFactory = new ConfigBucketOwnerProviderFactory();
         final BucketOwnerProvider bucketOwnerProvider = configBucketOwnerProviderFactory.createBucketOwnerProvider(s3SourceConfig);
 
-        final S3Service s3Service = new S3Service(s3SourceConfig, buffer, codec, pluginMetrics, bucketOwnerProvider);
+        Optional<S3SelectOptions> s3SelectOptional = Optional.ofNullable(s3SourceConfig.getS3SelectOptions());
+
+        final S3ObjectHandler s3Handler;
+		if (s3SelectOptional.isPresent()) {
+			s3Handler = new S3SelectObjectWorker(buffer, s3SourceConfig.getNumberOfRecordsToAccumulate(),
```

Comment:
Please create the async S3Client here. Then you can add it to the constructor. This follows from my comments above.